build_scripts/api/dreambooth/ui_functions.py

```
@@ -0,0 +1,1136 @@
+        tenc_lr=args.lora_txt_learning_rate,
+    )
+
+    sampler = BucketSampler(dataset, args.train_batch_size, True)
+    n_workers = 0
+
+    dataloader = torch.utils.data.DataLoader(
+        dataset,
+        batch_sampler=sampler,
+        batch_size=1,
+        shuffle=False,
+        drop_last=False,
+        collate_fn=debug_collate_fn,
+        pin_memory=True,
+        num_workers=n_workers,
+    )
+
+    lines = []
+    test_epochs = num_epochs
+    sim_train_steps = test_epochs * (dataloader.__len__() // batch_size)
+    print(
+        f"Simulating training for {test_epochs} epochs, batch size of {batch_size}, total steps {sim_train_steps}."
+    )
+    for epoch in mytqdm(range(test_epochs), desc="Simulating training.", position=0):
+        for step, batch in enumerate(dataloader):
+            image_names = batch["images"]
+            captions = batch["input_ids"]
+            losses = batch["loss_weights"]
+            last_lr = lr_scheduler.get_last_lr()[0]
+            line = f"Epoch: {epoch}, Batch: {step}, Images: {len(image_names)}, Loss: {losses.mean()} Last LR: {last_lr}"
+            print(line)
+            lines.append(line)
+            loss_idx = 0
+            for image, caption in zip(image_names, captions):
+                line = f"{image}, {caption}, {losses[loss_idx]}"
+                lines.append(line)
+                loss_idx += 1
+            lr_scheduler.step(args.train_batch_size)
+            optimizer.step()
+        lr_scheduler.step(1, is_epoch=True)
+    samples_dir = os.path.join(args.model_dir, "samples")
+    if not os.path.exists(samples_dir):
+        os.makedirs(samples_dir)
+    bucket_file = os.path.join(samples_dir, "prompts.json")
+    with open(bucket_file, "w") as outfile:
+        json.dump(lines, outfile, indent=4)
+    try:
+        del dataloader
+        del dataset.tokenizer
+        del dataset
+        del lr_scheduler
+        del optimizer
+        cleanup()
+    except:
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```